File: tests/testdata/programs/python/s3bucketmod/__main__.py
"""A Python Pulumi program"""

import pulumi
import pulumi_bucket as bucket

config = pulumi.Config()
prefix = config.get('prefix') or pulumi.get_stack()

testbucket = bucket.Module("test-bucket",
    bucket=f"{prefix}-test-bucket",

)

pulumi.export('bucketARN', testbucket.s3_bucket_arn)

File: tests/testdata/programs/go/s3bucketmod/main.go
package main

import (
	"github.com/pulumi/pulumi-terraform-module/sdks/bucket/v4"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {

	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		prefix := cfg.Get("prefix")
		if prefix == "" {
			prefix = ctx.Stack()
		}
		testbucket, err := bucket.NewModule(ctx, "test-bucket", &bucket.ModuleArgs{
			Bucket: pulumi.String(prefix + "-test-bucket"),
		})
		if err != nil {
			return err
		}

		ctx.Export("bucketARN", testbucket.S3_bucket_arn)
		return nil
	})
}
